eZ/Publish/Core/Persistence/Legacy/Content/Handler.php

```diff
@@ -587,20 +598,29 @@ public function listVersions($contentId, $status = null, $limit = -1)
      */
     public function copy($contentId, $versionNo = null)
     {
+        $contentInfo = $this->loadContentInfo($contentId);
+
         $currentVersionNo = isset($versionNo) ?
             $versionNo :
-            $this->loadContentInfo($contentId)->currentVersionNo;
+            $contentInfo->currentVersionNo;
 
         // Copy content in given version or current version
         $createStruct = $this->mapper->createCreateStructFromContent(
             $this->load($contentId, $currentVersionNo)
         );
+
+        if (!isset($versionNo) && $currentVersionNo > 1) {
+            $createStruct->initialLanguageId = $this
+                ->languageHandler
+                ->loadByLanguageCode($contentInfo->mainLanguageCode)
+                ->id;
+        }
```

Comment:
👍  Yes, and it already has language handler. So rather handle it correctly there straight away instead of having to fic the struct afterwards. @adamwojs Do you miss any kind of info there to be able to do that?